xds/src/main/java/io/grpc/xds/XdsLbState.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import io.grpc.Attributes;
+import io.grpc.ConnectivityStateInfo;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.LoadBalancer.Subchannel;
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * The states of an XDS working session of {@link XdsLoadBalancer}.  Created when XdsLoadBalancer
+ * switches to the current mode.  Shutdown and discarded when XdsLoadBalancer switches to another
+ * mode.
+ */
+abstract class XdsLbState {
```

Comment:
Does it have more than one implementations? There seems to be only one in XdsLoadBalancer. Can it be a concrete class instead?

Reply:
It will have two concrete class implementations, one for standard plugin, one for custom plugin. The current one in XdsLoadBalancer is temporary.